aiohttp/connector.py

```diff
@@ -181,6 +181,10 @@ def __init__(self, *, keepalive_timeout=sentinel,
         self._acquired_per_host = defaultdict(set)
         self._keepalive_timeout = keepalive_timeout
         self._force_close = force_close
+
+        # {host_key: FIFO list of waiters}
+        # NOTE: this is not a true FIFO because the true order is lost amongst
+        #       the dictionary keys
```

Comment:
I would suggest using the `deque` [1] data strucutre that is the one used by al of the implementations of `asyncio.locks`. More likely because lists have the following constraint:

> lists incur O(n) memory movement costs for pop(0)


[1] https://github.com/python/cpython/blob/master/Lib/asyncio/locks.py#L437